execution/events/tx.go

```diff
@@ -85,6 +91,31 @@ func SubscribeAccountOutputSendTx(ctx context.Context, subscribable event.Subscr
 	})
 }
 
+// Subscribe to all execution events generated by a particular Tx
+func SubscribeTx(ctx context.Context, subscribable event.Subscribable, txHash []byte, ch chan<- *Event) error {
+	qry := query.NewBuilder().AndEquals(event.TxHashKey, hex.EncodeUpperToString(txHash))
+
+	subscriber, err := event.GenerateSubscriptionID()
+	if err != nil {
+		return err
+	}
+
+	// Acts as a sentinel for the final event
+	confirmationEventID := EventStringTxConfirmation(txHash)
+
```

Comment:
Is it possible the tx will never make it to a block (consensus failure, another validator proposed block with tx for same address + same sequence number?) Do we leak stuff?